File: 217_contain_duplicate/contain_duplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector <int> & nums) {
        for (auto itr = nums.begin(); itr != nums.end(); itr++){
            if (unique[*itr] == 1) return true;
            unique[*itr] = 1;
        }
        return false;
    }
private:
    unordered_map <int, int> unique;
};

int main(void){
    vector <int> nums{3, 2, 5};

    Solution sol;
    cout << sol.containsDuplicate(nums) << "\n";
}

File: 0217_contain_duplicate/contain_duplicate.py
class Solution:
    def containsDuplicate(self, nums: list) -> bool:
        unique = {}
        for i in nums:
            try:
                unique[i]
                return True
            except KeyError:
                unique[i] = i
                
        return False
    
print(Solution.containsDuplicate(1, nums = [1, 2, 3, 1]))